sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -231,6 +252,24 @@ public ConfigurationClientBuilder connectionString(String connectionString) {
 
         this.endpoint = credential.getBaseUri();
 
+        // Make sure only use connection string
+        this.tokenCredential = null;
+        return this;
+    }
+
+    /**
+     * Sets the AAD credential to use when authenticating HTTP requests.
+     *
+     * @param tokenCredential AAD token
+     * @return The updated ConfigurationClientBuilder object.
+     * @throws NullPointerException If {@code credential} is {@code null}.
+     */
+    public ConfigurationClientBuilder credential(TokenCredential tokenCredential) {
+        Objects.requireNonNull(tokenCredential);
+        this.tokenCredential = tokenCredential;
+
+        // Make sure only use AAD token
+        this.credential = null;
```

Comment:
Why? You ensure this already in the .buildAsyncClient() part.

Reply:
I think it is better to null out the other credential when using the setter instead of the build method. In my opinion I would leave this as is and update the builder.